classes/search/pattern/class-patternusageapi.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * Table displaying blocks usage
+ *
+ * @package P4BKS\Search
+ */
+
+namespace P4GBKS\Search\Pattern;
+
+use WP_Block_Patterns_Registry;
+use P4GBKS\Search\Pattern\Query\Parameters;
+
+/**
+ * Pattern usage API
+ */
+class PatternUsageApi {
+
+	public const DEFAULT_POST_STATUS = [ 'publish' ];
+
+	/**
+	 * @var PatternUsage
+	 */
+	private $usage;
+
+	/**
+	 * @var Parameters
+	 */
+	private $params;
+
+	/**
+	 * @var array[] Blocks.
+	 */
+	private $items;
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		$this->usage  = new PatternUsage();
+		$this->params = ( new Parameters() )
+			->with_name( $this->pattern_names() )
+			->with_post_type( $this->allowed_post_types() )
+			->with_post_status( self::DEFAULT_POST_STATUS );
+	}
+
+	/**
+	 * Count patterns
+	 */
+	public function get_count(): array {
+		if ( null === $this->items ) {
+			$this->fetch_items();
+		}
+
+		$names    = $this->pattern_names();
+		$patterns = array_fill_keys( $names, [ 'total' => 0 ] );
+		ksort( $patterns );
+
+		foreach ( $this->items as $item ) {
+			$patterns[ $item['pattern_name'] ]['total']++;
+		}
+
+		return $patterns;
+	}
+
+	/**
+	 * Fetch parsed blocks
+	 */
+	private function fetch_items(): void {
+		$this->items = $this->usage->get_patterns( $this->params );
+	}
+
+	/**
+	 * Get all patterns names
+	 */
+	private function pattern_names(): array {
+		return array_filter(
+			array_column(
+				( WP_Block_Patterns_Registry::get_instance() )->get_all_registered(),
+				'name'
+			),
+			fn ( $name ) => 'p4/blank-page' !== $name
```

Comment:
Is this check should be applied to the `p4/blank-page` pattern or to all the pattern pages?

Reply:
@lithrel we've just renamed to [`p4/blank-page-pattern-layout`](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/891/files#diff-b38cca8eccfb381ce2f1d2dd175ad88062072c86eff8469108296637ba28229fR22).

Reply:
Updated using BlankPage::get_name() function instead of a string, so it stays up to date :grin: 
Also added a class to GetInformed so it's tracked